packages/react-query/src/shared/proxy/utilsProxy.ts

```diff
@@ -340,6 +365,10 @@ function createRecursiveUtilsProxy<TRouter extends AnyRouter>(
       },
       getData: () => context.getQueryData(queryKey),
       getInfiniteData: () => context.getInfiniteQueryData(queryKey),
+      setMutationDefaults: () =>
+        context.setMutationDefaults([queryKey[0]], input),
```

Comment:
what's this destructuring (`[queryKey[0]]`) needed for?

Reply:
This is directly connected to https://github.com/trpc/trpc/pull/5814#discussion_r1666738105. We can avoid it if we decide to use `TRPCQueryKey` as the mutation key in all these added functions.

Reply:
Replaced `[queryKey[0]]` with `getMutationKeyInternal(path)`.

`createRecursiveUtilsProxy` is still very "query-centric" (for good reasons). I think this is the best we can do without doing a bigger refactoring